java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -1721,15 +1729,47 @@ class DatabaseImpl(
                 CREATE INDEX number_primitive_value_index ON number_primitive_values (value);
             """.trimIndent().split("\n\n")
 
+        private val DROP =
+            """
+                DROP INDEX type_name_index;
+                DROP TABLE types;
+                DROP INDEX storage_key_index;
+                DROP TABLE storage_keys;
+                DROP TABLE entities;
+                DROP INDEX entity_refs_index;
+                DROP TABLE entity_refs;
+                DROP TABLE collections;
+                DROP INDEX collection_entries_collection_id_index;
+                DROP TABLE collection_entries;
+                DROP INDEX field_names_by_parent_type;
+                DROP TABLE fields;
+                DROP INDEX field_values_by_entity_storage_key;
+                DROP TABLE field_values;
+                DROP INDEX text_primitive_value_index;
+                DROP TABLE text_primitive_values;
+                DROP INDEX number_primitive_value_index;
+                DROP TABLE number_primitive_values;
+            """.trimIndent().split("\n")
+
         private val VERSION_2_MIGRATION = arrayOf("ALTER TABLE entities ADD COLUMN orphan INTEGER;")
 
-        private val MIGRATION_STEPS = mapOf(2 to VERSION_2_MIGRATION)
+        private val VERSION_3_MIGRATION = listOf(DROP, CREATE).flatten().toTypedArray()
+
+        private val MIGRATION_STEPS = mapOf(2 to VERSION_2_MIGRATION, 3 to VERSION_3_MIGRATION)
```

Comment:
have you tested this somehow? if so it would be good to explain how in the pr discussion for future reference.

Reply:
yeah, good point.

I tested by using the sqlite3 CLI as Jason suggested. I ran CREATE, then DROP, then manually verified that all tables and indexes were gon, then verified that CREATE could run again without errors.